microprofile-fault-tolerance/src/test/java/org/jboss/eap/qe/microprofile/fault/tolerance/v20/FaultTolerance20AsyncPartialTest.java

```
@@ -0,0 +1,232 @@
+        testPartial(20, baseApplicationUrl + "partial?operation=bulkhead15q5-timeout-retry&counter=", expectedResponses);
+    }
+
+    private static void testPartial(int parallelRequests, String url, Map<String, Range<Integer>> expectedResponses)
+            throws InterruptedException {
+        Set<String> violations = Collections.newSetFromMap(new ConcurrentHashMap<>());
+        Queue<String> seenResponses = new ConcurrentLinkedQueue<>();
+
+        ExecutorService executor = Executors.newFixedThreadPool(parallelRequests);
+        for (int i = 0; i < parallelRequests; i++) {
+            final int finalI = i;
+            executor.submit(() -> {
+                try {
+                    seenResponses.add(RestAssured.when().get(url + finalI).asString());
+                } catch (Exception e) {
+                    violations.add("Unexpected exception: " + e.getMessage());
+                }
+            });
+        }
+        executor.shutdown();
+        boolean finished = executor.awaitTermination(15, TimeUnit.SECONDS);
+        assertThat(finished, is(true));
+
+        for (String seenResponse : seenResponses) {
+            if (!expectedResponses.containsKey(seenResponse.replaceAll("[0-9]", ""))) {
+                violations.add("Unexpected response: " + seenResponse);
+            }
+        }
+        for (Map.Entry<String, Range<Integer>> expectedResponse : expectedResponses.entrySet()) {
+            int count = 0;
+            for (String seenResponse : seenResponses) {
+                if (expectedResponse.getKey().equals(seenResponse.replaceAll("[0-9]", ""))) {
+                    count++;
+                }
+            }
+            if (!expectedResponse.getValue().contains(count)) {
+                violations.add("Expected to see " + expectedResponse.getValue() + " occurrence(s) but seen " + count
+                        + ": " + expectedResponse.getKey());
+            }
+        }
+        assertThat(violations, is(empty()));
```

Same as previously - why not fail fast?

---

Same as above.